py/objstr.c

```diff
@@ -748,23 +768,11 @@ static mp_obj_t str_finder(size_t n_args, const mp_obj_t *args, int direction, b
 
     GET_STR_DATA_LEN(args[0], haystack, haystack_len);
```

Comment:
`haystack_len` is no longer used, so this could be maybe replaced with something slightly smaller/efficient?

Also, is `haystack` really needed now?  I guess it is to compute the correct offset for return value.